### PR TITLE
update API using ZAP 2.12.0

### PR DESCRIPTION
**Pull request by njmulsqb**
Signed-off-by: Najam Ul Saqib <najamulsaqib@tutamail.com>

Here's a PR that includes updated ZAP code. I am creating a PR just to add new features to the outdated NodeJS API.

I tried to update the generated files and followed steps from https://github.com/zaproxy/zap-api-python#updating-the-generated-files
Here are the steps I took:
1. Ran `./gradlew generateNodeJSApiEndpoints` in zaproxy directory
2. Added the newly added files to `index.js` following the pattern

If I am missing anything let me know.

### CODE REVIEW

**Comment by thc202** [comment]
The changelog should be updated. The add-on APIs should be regenerated as well.

**Comment by njmulsqb** [comment]
I tried to ran `generateNodeJsZapApiClientFiles` in zap-extensions but it is failing with an error, check this https://pastebin.com/2gTE7REq
As for changelog, what section needs to be changed and what version will this one be? 1.1?

**Comment by thc202** [comment]
Add the flag `--continue`. For the changelog, mention the update for 2.12 and the add-ons added/updated, e.g. https://github.com/zaproxy/zap-api-python/blob/master/CHANGELOG.md (the version should stay as Unreleased, that's changed during the release).

**Comment by njmulsqb** [comment]
I used the `--continue` with it, it showed lots of failure errors but it also added new files in the API, if you're interested here's the trace https://pastebin.com/xX50QymF

I have updated the changelog, cant really locate their versions so just mentioned the name, if version is required let me know where can I find it. 

I have created an account on NPM with username `njmulsqb` if thats required to update the NPM package. (I am amazed to see 3k+ weekly downloads on it)

**Comment by njmulsqb** [comment]
I dont know why it is asking me to pull each time I try to push, who is making the changes on remote repo? :P

**Comment by kingthorin** [comment]
It’s likely because you’re working on master. You should have created a new branch before starting your changes.

**Comment by njmulsqb** [comment]
@kingthorin Yeah, realized it very late in the process. Any workaround?

**Comment by kingthorin** [comment]
Don’t think so. You could just ignore the message.

**Comment by thc202** [comment]
The versions of the add-ons are in the corresponding `gradle.properties` files, e.g.: https://github.com/zaproxy/zap-extensions/blob/0d6fcd3f5f9fe90860e8ebd1d6d3b580756fd3bb/addOns/accessControl/gradle.properties Although the changelog is probably a better place to check the released version.

File: CHANGELOG.md [hunk 1]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
 
 ## [Unreleased]
 
+## [1.1.0] - 2022-11-26 - Upgrade API to ZAP 2.12.0
```
Comment (thc202) [inline CHANGELOG.md:8]: This should be removed.

File: CHANGELOG.md [hunk 2]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+
+### Added
+* Add the API of the following Add-Ons
+    * Access Control version 9
```
Comment (thc202) [inline CHANGELOG.md:12]: `8`

File: CHANGELOG.md [hunk 3]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Alert Filter version 15
```
Comment (thc202) [inline CHANGELOG.md:13]: `Alert Filters version 14`

File: CHANGELOG.md [hunk 4]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Automation version 0.20.0
```
Comment (thc202) [inline CHANGELOG.md:14]: `Automation Framework version 0.19.0`

File: CHANGELOG.md [hunk 5]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Exim version 0.4.0
```
Comment (thc202) [inline CHANGELOG.md:15]: `Import/Export version 0.3.0`

File: CHANGELOG.md [hunk 6]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * GraphQL version 0.13.0
```
Comment (thc202) [inline CHANGELOG.md:16]: `GraphQL Support version 0.12.0`

File: CHANGELOG.md [hunk 7]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Network version 0.6.0
```
Comment (thc202) [inline CHANGELOG.md:17]: `0.5.0`

File: CHANGELOG.md [hunk 8]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Reports version 0.18.0
```
Comment (thc202) [inline CHANGELOG.md:18]: `Report Generation version 0.17.0`

File: CHANGELOG.md [hunk 9]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Retest version 0.5.0
```
Comment (thc202) [inline CHANGELOG.md:19]: `0.4.0`

File: CHANGELOG.md [hunk 10]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Revisit version 5
```
Comment (thc202) [inline CHANGELOG.md:20]: `4`

File: CHANGELOG.md [hunk 11]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Wappalyzer version 21.17.0
```
Comment (thc202) [inline CHANGELOG.md:21]: `Wappalyzer - Technology Detection version 21.16.0`

File: CHANGELOG.md [hunk 12]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+
+### Changed
+* Update Core APIs for 2.12.0
+* Update the APIs of following Add-Ons
+    * Ajax Spider version 23.11.0
```
Comment (thc202) [inline CHANGELOG.md:26]: `23.10.0`

File: CHANGELOG.md [hunk 13]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * OpenAPI version 31
```
Comment (thc202) [inline CHANGELOG.md:27]: `OpenAPI Support version 30`

File: CHANGELOG.md [hunk 14]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * PNH version 14
```
Comment (thc202) [inline CHANGELOG.md:28]: `Plug-n-Hack Configuration version 13`

File: CHANGELOG.md [hunk 15]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Replacer version 12
```
Comment (thc202) [inline CHANGELOG.md:29]: `11`

File: CHANGELOG.md [hunk 16]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Reveal version 6
```
Comment (thc202) [inline CHANGELOG.md:30]: `5`

File: CHANGELOG.md [hunk 17]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Selenium version 15.12.0
```
Comment (thc202) [inline CHANGELOG.md:31]: `15.11.0`

File: CHANGELOG.md [hunk 18]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * SOAP version 17
```
Comment (thc202) [inline CHANGELOG.md:32]: `SOAP Support version 16`

File: CHANGELOG.md [hunk 19]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Spider version 0.2.0
```
Comment (thc202) [inline CHANGELOG.md:33]: `0.1.0`

File: CHANGELOG.md [hunk 20]
```diff
@@ -5,6 +5,33 @@ The format is based on [Keep a Changelog](http://keepachangelog.com/en/1.0.0/).
...
+    * Web Socket version 28
```
Comment (thc202) [inline CHANGELOG.md:34]: `WebSockets version 27`

**Comment by njmulsqb** [comment]
Versions corrected!

**Comment by njmulsqb** [comment]
When is it expected to merge?

**Comment by thc202** [comment]
I guess after the merge of the other PR, no point merging this if it's going to be rewritten.

**Comment by njmulsqb** [comment]
Agreed, 2.12 should be only generated now once all the breaking changes are implemented, like the PR you mentioned and the HTTP library one.

**Pull request closed without merging**